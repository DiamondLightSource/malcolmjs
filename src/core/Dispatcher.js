/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
