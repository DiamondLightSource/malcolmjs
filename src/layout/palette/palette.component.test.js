import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import configureStore from 'redux-mock-store';
import Palette from './palette.component';

describe('Palette', () => {
  let shallow;
  const mockStore = configureStore();
  let state;

  beforeEach(() => {
    shallow = createShallow({
      dive: true,
    });

    state = {
      malcolm: {
        blocks: {
          parent: {
            attributes: [
              {
                calculated: {
                  name: 'layout',
                  layout: {
                    blocks: [
                      { name: 'block 1', visible: true },
                      { name: 'block 2', mri: 'panda:block2', visible: false },
                    ],
                  },
                },
              },
            ],
          },
        },
        parentBlock: 'parent',
        mainAttribute: 'layout',
      },
    };
  });

  it('renders correctly', () => {
    const tree = shallow(<Palette store={mockStore(state)} />);
    expect(tree.dive()).toMatchSnapshot();
  });
});
