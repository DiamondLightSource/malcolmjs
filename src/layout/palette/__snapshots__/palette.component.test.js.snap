// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette renders correctly 1`] = `
<div
  className="Palette-container-1"
>
  Palette goes here
</div>
`;
